clients/js/src/embeddings/CohereEmbeddingFunction.ts

```diff
@@ -0,0 +1,31 @@
+import { IEmbeddingFunction } from "./IEmbeddingFunction";
+
+let CohereAiApi: any;
+
+export class CohereEmbeddingFunction implements IEmbeddingFunction {
+    private api_key: string;
+    private model: string;
+
+    constructor({ cohere_api_key, model }: { cohere_api_key: string, model?: string }) {
+        try {
+            // eslint-disable-next-line global-require,import/no-extraneous-dependencies
+            CohereAiApi = require("cohere-ai");
+        } catch {
+            throw new Error(
+                "Please install the cohere-ai package to use the CohereEmbeddingFunction, `npm install -S cohere-ai`"
+            );
+        }
+        this.api_key = cohere_api_key;
+        this.model = model || "large";
+    }
+
+    public async generate(texts: string[]) {
+        const cohere = CohereAiApi.init(this.api_key);
```

Comment:
not your mistake, but can we remove these 2 unsused lines? 

```
const cohere = CohereAiApi.init(this.api_key);
const embeddings = [];
```

don't appear to be used at all